horace_core/spaghetti_plot.m

```diff
@@ -186,17 +186,17 @@
 
 qbin = opt.qbin;
 qwidth = opt.qwidth;
+szq = numel(qwidth(:));
 
-switch size(opt.qwidth)
-  case [1, 1]                 % "Square" for all segments
-    qwidth = repmat(opt.qwidth, 2, nseg);
-  case {[1, 2], [2, 1]}       % Rectangular for all segments
-    qwidth = repmat(opt.qwidth(:)', 1, nseg);
-  case {[nseg, 1], [1, nseg]} % "Square" for each segment
+if szq == 1 % "Square" for all segments
+    qwidth = repmat(opt.qwidth, 2, nseg);  
+elseif szq == 2 % Rectangular for all segments
+    qwidth = repmat(opt.qwidth(:), 1, nseg);
+elseif szq == nseg  % "Square" for each segment
     qwidth = repmat(opt.qwidth(:), 2, 1);
-  case [2, nseg]              % Rectangular for each segments
+elseif szq == 2*nseg % Rectangular for each segments
     qwidth = reshape(opt.qwidth, 2, nseg);
-  otherwise
+else
     error('HORACE:spaghetti_plot:invalid_argument', ...
           ['qwidth size must be one of: [1, 1], [2, 1], [1, nseg], or [2, nseg].\n' ...
            'Received: %s'], disp2str(opt.qwidth))
```

Comment:
This is fundamentally wrong and will cause further errors
```suggestion
siz = size(opt.qwidth);

if isequal(siz, [1, 1])                % "Square" for all segments
    qwidth = repmat(opt.qwidth, 2, nseg);
elseif isequal(siz,  [1, 2]) || ...    % Rectangular for all segments
        isequal(siz, [2, 1])
    qwidth = repmat(opt.qwidth(:), 1, nseg);
elseif isequal(siz,  [nseg, 1]) || ... % "Square" for each segment
        isequal(siz, [1, nseg])
    qwidth = repmat(opt.qwidth(:)', 2, 1);
elseif isequal(siz,  [2, nseg])        % Rectangular for each segments
    qwidth = reshape(opt.qwidth, 2, nseg);
else
    error('HORACE:spaghetti_plot:invalid_argument', ...
          ['qwidth size must be one of: [1, 1], [2, 1], [1, nseg], or [2, nseg].\n' ...
           'Received: %s'], disp2str(opt.qwidth))
```

Reply:
Have not found any errors, so do not see what is wrong with this. The suggested change fails with error
```SWITCH expression must be a scalar or a character vector.

Error in spaghetti_plot (line 190)
```
on the version of the Matlab installed in IDAAAS

Reply:
There is a significant difference between an array with 12 elements and a 4x3 array. Particularly where repmat is concerned.

Reply:
The suggested change has a blank line on line 190, so I am not sure what you have copied